a False Positives example

```bash
$ go run -v main.go
Found existing alias for "go run". You should use: "gor"
command-line-arguments
"level"=0 "msg"="default info log" "stringVal"="value" "intVal"=12345
"level"=0 "msg"="default info log" "key1"="<no-value>"
"level"=0 "msg"="default info log" "key1"="val1"
$ logrlint ./...
/Users/fupeng/go/src/tmpa/main.go:26:29: odd number of arguments passed as key-value pairs for logging
/Users/fupeng/go/src/tmpa/main.go:28:29: odd number of arguments passed as key-value pairs for logging
```

main.go
```go
// https://github.com/go-logr/logr/blob/master/example_test.go
package main

import (
	"fmt"

	"github.com/go-logr/logr"
	"github.com/go-logr/logr/funcr"
)

// NewStdoutLogger returns a logr.Logger that prints to stdout.
func NewStdoutLogger() logr.Logger {
	return funcr.New(func(prefix, args string) {
		if prefix != "" {
			_ = fmt.Sprintf("%s: %s\n", prefix, args)
		} else {
			fmt.Println(args)
		}
	}, funcr.Options{})
}

func main() {
	l := NewStdoutLogger()
	l.Info("default info log", "stringVal", "value", "intVal", 12345)
	t := make([]interface{}, 0)
	t = append(t, "key1")
	l.Info("default info log", t...)  // should check
	t = append(t, "val1")
	l.Info("default info log", t...)  // False Positives here ! 
	// Output:
	// "level"=0 "msg"="default info log" "stringVal"="value" "intVal"=12345
	// "level"=0 "msg"="V(0) info log" "stringVal"="value" "intVal"=12345
	// "msg"="error log" "error"="an error" "stringVal"="value" "intVal"=12345
}

```

--- a/logrlint.go
+++ b/logrlint.go
@@ -64,6 +64,11 @@ func checkEvenArguments(pass *analysis.Pass, call *ast.CallExpr) {
 		return
 	}
 
+	// ellipsis args is hard, just skip
+	if call.Ellipsis.IsValid() {
+		return
+	}
+
 	params := sig.Params()
 	nparams := params.Len() // variadic => nonzero
 	args := params.At(nparams - 1)

--- a/testdata/src/a/a.go
+++ b/testdata/src/a/a.go
@@ -8,16 +8,19 @@ import (
 )
 
 func Example() {
+	err := fmt.Errorf("error")
 	log := logr.Discard()
-	log = log.WithValues("key")                                         // want `odd number of arguments passed as key-value pairs for logging`
-	log.Info("message", "key1", "value1", "key2", "value2", "key3")     // want `odd number of arguments passed as key-value pairs for logging`
-	log.Error(fmt.Errorf("error"), "message", "key1", "value1", "key2") // want `odd number of arguments passed as key-value pairs for logging`
-	log.Error(fmt.Errorf("error"), "message", "key1", "value1", "key2", "value2")
+	log = log.WithValues("key")                                     // want `odd number of arguments passed as key-value pairs for logging`
+	log.Info("message", "key1", "value1", "key2", "value2", "key3") // want `odd number of arguments passed as key-value pairs for logging`
+	log.Error(err, "message", "key1", "value1", "key2")             // want `odd number of arguments passed as key-value pairs for logging`
+	log.Error(err, "message", "key1", "value1", "key2", "value2")
 
 	var log2 logr.Logger
 	log2 = log
 	log2.Info("message", "key1") // want `odd number of arguments passed as key-value pairs for logging`
 
 	log3 := logr.FromContextOrDiscard(context.TODO())
-	log3.Error(fmt.Errorf("error"), "message", "key1") // want `odd number of arguments passed as key-value pairs for logging`
+	log3.Error(err, "message", "key1") // want `odd number of arguments passed as key-value pairs for logging`
+	args := []interface{}{"abc"}
+	log3.Error(err, "message", args...)
 }

--- a/testdata/src/a/go.mod
+++ b/testdata/src/a/go.mod
@@ -1,5 +1,5 @@
 module a
 
-go 1.19
+go 1.16
 
 require github.com/go-logr/logr v1.2.3

